kernel/USER_GUIDE.md

```diff
@@ -91,57 +91,85 @@ while(fileIter.hasNext()) {
+            }
+          }
+
+	  // access the data for column at ordinal 1
+	  ColumnVector column1 = dataBatch.getColumnVector(1);
+	  for (int rowIndex = 0; rowIndex < column1.getSize(); rowIndex++) {
+            // check if the row is selected or not
+            if (selectionVector.isEmpty() || selectionVector.get().getBoolean(rowIndex)) {
+              // Assuming the column type is Long.
+              // If it is a different type, call the relevant function on the `ColumnVector`
+              System.out.println(column1.getLong(rowIndex));
+            }
+          }
+	  // .. more ..
+        }
       }
     }
-
-    // access the data for column at ordinal 1
-    ColumnVector column1 = dataBatch.getColumnVector(1);
-    for (int rowIndex = 0; rowIndex < column1.getSize(); rowIndex++) {
-      // check if the row is selected or not
-      if (selectionVector.isEmpty() || selectionVector.get().getBoolean(rowIndex)) {
-        // Assuming the column type is Long.
-        // If it is a different type, call the relevant function on the `ColumnVector`
-        System.out.println(column1.getLong(rowIndex));
-      }
-    }
-	// .. more ..
   }
 }
 ```
 
 A few working examples to read Delta tables within a single process are available [here](https://github.com/delta-io/delta/tree/master/kernel/examples).
 
 #### Important Note
-* All the Delta protocol-level details are encoded in the rows returned by `Scan.getScanFiles` API, but you do not have to understand them in order to read the table data correctly. You can mostly treat these rows as opaque objects and pass them correctly to the `Scan.readData()` method. In the future, there will be more information (that is, more columns) added to these rows, but your code will not have to change to accommodate those changes. This is the major advantage of the abstractions provided by Delta Kernel.
+* All the Delta protocol-level details are encoded in the rows returned by `Scan.getScanFiles` API, but you do not have to understand them in order to read the table data correctly. All you need is to get the physical Parquet file details from the scan file row, read the data from the Parquet file, and return in `ColumnarBatch` format. The physical data can be converted into the data in the logical data of the table using [`Scan.transformPhysicalData`](https://delta-io.github.io/delta/snapshot/kernel-api/java/io/delta/kernel/Scan.html#transformPhysicalData-io.delta.kernel.client.TableClient-io.delta.kernel.data.Row-io.delta.kernel.data.Row-io.delta.kernel.utils.CloseableIterator-). Transformation to logical data involves adding partition column data, deleting rows that are marked as deleted by the scan file deletion vector. In the future, there will be more information (i.e. more columns) added to these logical data returned, but your code will not have to change to accommodate those changes. This is the major advantage of the abstractions provided by Delta Kernel.
```

Comment:
maybe "... get the physical Parquet file status from the scan file row" to keep our wording consistent?